lix/packages/client/src/repoState.ts

```diff
@@ -103,41 +230,68 @@ export async function repoState(
 			throw new Error("fs provider does not support placeholders")
 		}
 		console.info("Using lix for cloning repo")
-
+		// FIXME: symlink = false in git/config!???
 		await isoGit
 			.clone({
-				fs: withProxy({ nodeishFs: rawFs, verbose: debug, description: "clone" }),
+				fs: rawFs,
 				http: makeHttpClient({
 					debug,
 					description: "clone",
-
-					onReq: ({ url, body }: { url: string; body: any }) => {
-						return optimizedRefsReq({ url, body, addRef: state.branchName })
-					},
-
-					onRes: optimizedRefsRes,
+					onReq: experimentalFeatures.lazyClone
+						? optimizeReq.bind(null, {
+								noBlobs: true,
+								addRefs: [state.branchName || "HEAD"],
+						  })
+						: undefined,
+					onRes: experimentalFeatures.lazyClone ? optimizeRes : undefined,
 				}),
 				dir,
 				cache,
 				corsProxy: gitProxyUrl,
 				url: gitUrl,
-				singleBranch: true,
+				singleBranch: false, // if we clone with single branch true we will not get the defatult branch set in isogit
 				noCheckout: experimentalFeatures.lazyClone,
 				ref: state.branchName,
 
 				// TODO: use only first and last commit in lazy clone? (we need first commit for repo id)
```

Comment:
is this todo still relevant?

Reply:
yes, we still clone the whole repo history just to do project ids , we should find a way to get this from the git protocol directly